libs/util/http.client.js

```diff
@@ -211,6 +211,9 @@ function io(url, options) {
         resp.responseText = body;
 
         if (err) {
+            if (err.message === 'XMLHttpRequest timeout') {
+                err = new Error('XMLHttpRequest timeout: failed to load ' + resp.url + ' in ' + options.timeout + 'ms');
+            }
```

Comment:
Ah.  Good catch.  We can always set `err.timeout` to `options.timeout` (same way we set `err.url`).  It should just be debug information any ways.